include/xsparse/level_capabilities/co_iteration.hpp

```diff
@@ -6,8 +6,40 @@
 #include <algorithm>
 #include <stdexcept>
 
+#include <xsparse/level_capabilities/locate.hpp>
+
 namespace xsparse::level_capabilities
 {
+    /*
+    The class template for Coiteration of level formats.
+
+    Uses a generic function object F to compare elements
+    from different sequences at the same position and returns a tuple of the
+    minimum index and the corresponding elements from each sequence.
+
+    Parameters
+    ----------
+    F : class
+        A function object that is used to compare two elements from different ranges.
+    IK : class
+        The type of the first element of each range.
+    PK : class
+        The type of the second element of each range.
+    Levels : Tuple of class
+        A tuple of level formats, where each level is itself a tuple of elements to be iterated.
+    Is : Tuple of class
+        A tuple of indices that is used to keep track of the current position in each level.
+
+    Notes
+    -----
+    Coiteration is only allowed through tuples of levels if the following criterion is met:
+    If:
+        1. the levels are all ordered (i.e. has the `is_ordered == True` property)
+        2. if any of the level are do not have the is_ordered property, it must have the locate
+            function, else return False. Then do a check that `m_comparisonHelper` defines
+            a conjunctive merge (i.e. AND operation).
+    Otherwise, coiteration is not allowed.
+    */
```

Comment:
This is great! However, I'd recommend Doxygen-style docstrings.

Reply:
Okay, I changed the docstring to this style: https://www.doxygen.nl/manual/docblocks.html. Not as familiar w/ Doxygen, so lmk if you think I did anything wrong here.